tls/s2n_config.c

```diff
@@ -808,6 +812,30 @@ int s2n_config_set_extension_data(struct s2n_config *config, s2n_tls_extension_t
     return 0;
 }
 
+int s2n_config_set_custom_critical_oids(struct s2n_config *config, const char *const *custom_critical_oids, uint32_t custom_oid_count)
+{
+    POSIX_ENSURE_REF(config);
+
+#ifndef S2N_LIBCRYPTO_SUPPORTS_CUSTOM_OID
+    POSIX_BAIL(S2N_ERR_INTERNAL_LIBCRYPTO_ERROR);
+#endif
```

Comment:
Branching with ifdefs makes the code a bit harder to follow, and it's not actually necessary here since you're not branching on anything that requires the `S2N_LIBCRYPTO_SUPPORTS_CUSTOM_OID` feature in order to compile.

Instead, I'd use a helper function:
```
bool s2n_libcrypto_supports_custom_oid() {
#if S2N_LIBCRYPTO_SUPPORTS_CUSTOM_OID
    return true;
#else
    return false;
#endif
}
```
```suggestion
if s2n_libcrypto_supports_custom_oid() {
    POSIX_BAIL(S2N_ERR_INTERNAL_LIBCRYPTO_ERROR);
}
```